scipy/optimize/_minimize_constrained.py

```diff
@@ -0,0 +1,485 @@
+                           penalty, opt, c_viol):
+        print("|{0:>7}|{1:>7}|{2:>7}|   {3:^1.2e}  | "
+              "{4:^1.2e} | {5:^1.2e} | {6:^1.2e} | {7:^1.2e} |"
+              .format(niter, nfev, cg_niter, barrier_parameter,
+                      tr_radius, penalty, opt, c_viol))
+
+    @staticmethod
+    def print_footer():
+        print("")
+        print((7*3 + 13 + 10*4 + 9)*"-")
+        print("")
+
+
+def minimize_constrained(fun, x0, grad, hess=None, constraints=(),
+                         method=None, xtol=1e-8, gtol=1e-8,
+                         sparse_jacobian=None, options={},
+                         callback=None, max_iter=1000,
+                         verbose=0):
+    """Minimize scalar function subject to constraints.
+
+    Parameters
+    ----------
+    fun : callable
+        The objective function to be minimized.
+
+            fun(x) -> float
+
+        where x is an array with shape (n,).
+    grad : callable
+        Gradient of the objective function:
+
+            grad(x) -> array_like, shape (n,)
+
+        where x is an array with shape (n,).
+    hess : {callable, None}, optional
+        Hessian of the objective function:
+
+            hess(x) -> {LinearOperator, sparse matrix,  ndarray}, shape (n, n)
+
+        where x is an array with shape (n,).
+    x0 : ndarray, shape (n,)
+        Initial guess. ``len(x0)`` is the dimensionality of the minimization
+        problem.
+    method : {str, None}, optional
+        Type of solver. Should be on of:
+
+            - 'equality-constrained-sqp'
+            - 'tr-interior-point'
+
+        When ``None`` the more appropriate method is choosen.
+        'equality-constrained-sqp' is chosen for problems that
+        only have equality constraints and 'tr-interior-point'
+        for general optimization problems.
+    constraints : Constraint or List of Constraint's, optional
+        A single object or a list of objects specifying
+        constraints to the optimization problem.
+        Available constraints are:
+
+            - `BoxConstraint`
+            - `LinearConstraint`
+            - `NonlinearConstraint`
+
+    xtol : float, optional
+        Tolerance for termination by the change of the independent variable.
+        The algorithm will terminate when ``delta < xtol``, where ``delta``
+        is the algorithm trust-radius. Default is 1e-8.
+    gtol : float, optional
+        Tolerance for termination by the norm of the lagrangian gradient.
+        The algorithm will terminate when both the infinite norm of the
+        lagrangian gradient and the constraint violation are smaller than
+        ``gtol``. Default is 1e-8.
+    sparse_jacobian : {boolean, None}
+        The algorithm uses a sparse representation of the Jacobian if True
+        and a dense representation if False. When ``None`` the algorithm
+        uses the more convenient option (which is not always the more effective
```

Comment:
That explanation is indeed definitely very vague, I will try to improve it. Something like is probably better:

        The algorithm uses a sparse representation of the Jacobian if True
        and a dense representation if False. When `sparse_jacobian` is None
        the choice is done internally by the algorithm, using a sparse
        representation if at least one of the constraint Jacobians are sparse
        and a dense representation when they are all dense arrays.

I removed the phrase ``(which is not always the more effective)`` because it only make things more confusing and does not actually provide a solution. The default option ``None``, generally provide an efficient option.